### ISSUE TITLE
YOLO26 Errors exporting TensorRT

### ISSUE THREAD

**Issue by golden452** [ultralytics/ultralytics#23756]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

YOLO26 model throws warning whenever exporting with `end2end = True`. I tied both trained-from-scratch models and official pretrained YOLO26 models. 

When `end2end` is set to `False`, no warnings are raised. I assume it's something with the end2end mode. I do want to keep end2end mode on because it increases inference speed. 

After this warning, it still produces a usable .engine trt model. But I wonder if there are any potential risks ignoring this error. 

Can we safely ignore this warning or is there anything to do to fix this issue?

OS:

>Ubuntu 24.04 Ultralytics 8.4.18 🚀 Python-3.12.12 torch-2.10.0+cu130 CUDA:0 (NVIDIA GeForce RTX 5070 Ti, 15834MiB)

code: 
```
from ultralytics import YOLO

PT_PATH ='yolo26x.pt' 
model = YOLO(PT_PATH)
model.export(format="engine",end2end=True)
```

Errors:
> [/home/user/.conda/envs/sahi/lib/python3.12/site-packages/torch/onnx/_internal/torchscript_exporter/symbolic_opset11.py:954](http://localhost:8888/home/user/.conda/envs/sahi/lib/python3.12/site-packages/torch/onnx/_internal/torchscript_exporter/symbolic_opset11.py#line=953): UserWarning: Exporting aten::index operator of advanced indexing in opset 20 is achieved by combination of multiple ONNX operators, including Reshape, Transpose, Concat, and Gather. If indices include negative values, the exported graph will produce incorrect results.

> return opset9.index(g, self, index)

Thanks for any input. 

### Environment

Ultralytics 8.4.18 🚀 Python-3.12.12 torch-2.10.0+cu130 CUDA:0 (NVIDIA GeForce RTX 5070 Ti, 15834MiB)

OS                     Linux-6.17.0-14-generic-x86_64-with-glibc2.39
Environment            Linux
Python                 3.12.12
Install                pip
Path                   /home/user/.conda/envs/sahi/lib/python3.12/site-packages/ultralytics

GPU count              1
CUDA                   13.0

numpy                  ✅ 2.4.2>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.13.0.92>=4.6.0
pillow                 ✅ 12.1.1>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.17.1>=1.4.1
torch                  ✅ 2.10.0+cu130>=1.8.0
torch                  ✅ 2.10.0+cu130!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.25.0+cu130>=0.9.0
psutil                 ✅ 7.2.2>=5.8.0
polars                 ✅ 1.38.1>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18

### Minimal Reproducible Example

from ultralytics import YOLO

PT_PATH ='yolo26x.pt' 
model = YOLO(PT_PATH)
model.export(format="engine",end2end=True)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @golden452, thanks for opening this report and for using Ultralytics 🚀 This is an automated response to help triage the issue quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can ignore the warning. It doesn't affect the model.
